coordinate.go

```diff
@@ -418,3 +420,36 @@ func pingNode(addr string, method string) (time.Duration, error) {
 		return 0, pingErr
 	}
 }
+
+// Needed for cases of unregister->reregister, so the newly re-registered
+// node doesn't use the old status (which would stay for timed duration).
+// Also clear an inflightPings flag if present.
+func (a *Agent) checkNodeTracking(inCh nodeChannel) nodeChannel {
+	outCh := make(chan []*api.Node)
+	oldNodes := []*api.Node{}
+	go func() {
+		for nodes := range inCh {
+			inUse := make(map[string]bool, len(oldNodes))
```

Comment:
This might not necessarily warrant a change, but do we typically use bool-maps for sets? I've seen a few go projects use the empty struct instead to avoid problems with future authors assigning `false` into the map.

Reply:
If this were a longer lived map, on the struct or something I might agree. IMO for local scoped use cases misuse isn't really a risk and a boolean reads better.